eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -179,7 +179,7 @@ public function removeURLAliases(array $urlAliases)
     public function lookup($url)
     {
         $cacheItem = $this->cache->getItem(
-            'ez-urlAlias-url-' . str_replace(['/', ':', '(', ')', '@'], ['_S', '_C', '_B', '_B', '_A'], $url)
+            'ez-urlAlias-url-' . str_replace(['/', ':', '(', ')', '@', '\\', '{', '}'], ['_S', '_C', '_B', '_B', '_A', '_BS', '_CB', '_CB'], $url)
```

Comment:
q: does this not lead to the risk of collisions, given  that _BS is a valid token in an url? If we use '_' as prefix for replacement tokens we should at least repalce all '_' in existing string with '__'

Reply:
ps github srews my undescores

Reply:
I think that probability is very low, but indeed, it potentially may lead to the collision. 
Anyway, your comment applies to all of the tokens if we count it. WDYT @alongosz / @andrerom? 

> we use '' 

Do you mean `'\\'`? It's replacement for `\`.

Reply:
At a minimum doing a `_` => `__` as first replacement here would make a lot of sense indeed.
And maybe also make sure we have covered all restricted characters for Cache here.

Reply:
@gggeek / @andrerom changed in https://github.com/ezsystems/ezpublish-kernel/pull/2589/commits/2ab15cc03cb8cb1bb2d7422ad2259cc23594f920.

Reply:
@kmadejski regardless of probability, and of importance of conflicts for this very specific cache pool, my point is: the principle of token replacement when dealing with cache keys, and cache-collision in general, should be well understood by anyone who deals with the topic. Cache collisions are horrible to debug, as they are almost impossible to reproduce, and a very nice vector for security exploits. So better nip it in the bud.
Thx for listening :-)

Reply:
@gggeek yes, I totally agree, it's extremely hard to debug such bugs later on. Therefore, thanks for pointing this out 👍

Reply:
ps: sorry, I spoke too soon. It seems your fixup commit misses the point... Please allow me to explain:

whether you use one or 2 underscores as part of replacement token does not matter. The fact is that 'xxBS' is valid as part of initial key. So 'xxBS}' and '}xxBS' source keys will lead both to same key 'xxBSxxBS' . Collision.

What you want is to transform by using a token outside of the realm of possible values in the initial key.
As Andre said, this is easy to achieve with transform every x char into an xx char:
x => xx
xx => xxxx
x} =>xBs
xBs => xxBs
etc... you can try it out 4 yourself: there is no chance of collision. Maths.

Reply:
last comment 4 today (hopefully): also please use different substitution tokens for opening and closing parenthesis chars...

Reply:
Indeed, @kmadejski what I meant to be explity is:

```suggestion
            'ez-urlAlias-url-' . str_replace(['_'. '/', ':', '(', ')', '@', '\\', '{', '}'], ['__', '_S', '_C', '_B', '_B', '_A', '_BS', '_CB', '_CB'], $url)
```

Reply:
@gggeek / @andrerom thank you guys for the explanation. 
Indeed, I've missed the point here 🤦‍♂️

Anyway, fixed in https://github.com/ezsystems/ezpublish-kernel/pull/2589/commits/06ccff00b5e409dcb99b47f2e1fc2b05ebc55293

Reply:
great. +1